src/developer/functions/https/createKey.js

```diff
@@ -149,3 +157,43 @@ export const generateRegexFromPattern = (inputPattern) => {
 
   return pattern;
 };
+
+/**
+ * Checks to see if the account has a key with the same pattern and mode
+ * @param {[]} keys - The keys for the account
+ * @param {{
+ *  for: string,
+ *  pattern: string,
+ *  ip: string,
+ *  type: 'browser' | 'server',
+ *  mode: 'development' | 'production',
+ *  notes: string
+ * }} data - The forms data
+ * @returns {Promise<string?>} The duplicate API key or null
+ */
+export const getDuplicateKey = (keys, data) => {
+  let matchingKey = null;
+
+  if (!keys || keys?.length === 0 || !Array.isArray(keys)) {
+    return matchingKey;
+  }
+
+  for (const key of keys) {
+    // if the modes are different they are not the same key
+    if (key?.flags?.production !== (data.mode === 'production')) {
+      continue;
+    }
+
+    // if the key's pattern or ip matches
+    if (
+      (key?.flags?.server === false && key?.pattern === data.pattern) ||
+      (key?.flags?.server && key?.pattern === data.ip)
+    ) {
```

Comment:
```suggestion
    if (key?.pattern === (key?.flags?.server) ? data.ip : data.pattern) {
```

Slightly more simple way to do it...